Jantimizer/QueryOptimiserTest/QueryOptimiserTest.cs

```diff
@@ -15,63 +15,113 @@ public class QueryOptimiserTest
+    new int[] { 0, 0, 0 },                                                                              // min
+    new int[] { 150, 100, 50 },                                                                         // max
+    new int[] { 1, 0 },                                                                                 // nodeId
+    new ComparisonType.Type[] { ComparisonType.Type.More, ComparisonType.Type.More },                   // nodeComparisonType
+    new string[] { "A", "B" },                                                                          // nodeLeftTable
+    new string[] { "B", "C" },                                                                          // nodeRightTable
+    new string[] { "ID", "ID" },                                                                        // nodeLeftAttribute
+    new string[] { "ID", "ID" },                                                                        // nodeRightAttribute
+    new string[] { "A > B", "B > C" }                                                                   // nodeCondition
+    )]
+    [DataRow(
+    4,                                                                                                  // testID
+    new int[] { 0, 1 },                                                                                 // expectedNodeOrder
+    new string[] { "A", "B", "C" },                                                                     // name
+    new int[] { 10, 10, 10 },                                                                           // depth
+    new int[] { 0, 0, 0 },                                                                              // min
+    new int[] { 50, 100, 150 },                                                                         // max
+    new int[] { 1, 0 },                                                                                 // nodeId
+    new ComparisonType.Type[] { ComparisonType.Type.Less, ComparisonType.Type.Less },                   // nodeComparisonType
+    new string[] { "B", "A" },                                                                          // nodeLeftTable
+    new string[] { "C", "B" },                                                                          // nodeRightTable
+    new string[] { "ID", "ID" },                                                                        // nodeLeftAttribute
+    new string[] { "ID", "ID" },                                                                        // nodeRightAttribute
+    new string[] { "B < C", "A < B" }                                                                   // nodeCondition
+    )]
+
+    public void OptimiseJoinQueryEqual(int testID, int[] expectedNodeOrder, string[] name, int[] depth, int[] min, int[] max, int[] nodeId, ComparisonType.Type[] nodeComparisonType, string[] nodeLeftTable, string[] nodeRightTable, string[] nodeLeftAttribute, string[] nodeRightAttribute, string[] nodeCondition)
```

Comment:
For the love of god, please use [DynamicData](https://makolyte.com/csharp-use-dynamicdata-attribute-to-pass-functions-and-objects-into-parameterized-tests/) instead of this nightmare!